samples/algorithms/integer-factorization/Program.cs

```diff
@@ -82,15 +87,16 @@ static int Simulate(SimulateOptions options)
                     // Make sure to use simulator within using block. 
                     // This ensures that all resources used by QuantumSimulator
                     // are properly released if the algorithm fails and throws an exception.
-                    using (QuantumSimulator sim = new QuantumSimulator())
+                    using (var sim = options.UseDense ? (CommonNativeSimulator)new QuantumSimulator() : new SparseSimulator())
```

Comment:
It is suspicious that only the QuantumSimulator instance is cast to CommonNativeSimulator, but the SparseSimulator instance is not. I expected both cast.
```suggestion
                    using (var sim = options.UseDense ? (CommonNativeSimulator)new QuantumSimulator() : (CommonNativeSimulator)new SparseSimulator())
```